atc/worker/container_spec.go

```diff
@@ -14,10 +14,12 @@ type WorkerSpec struct {
 	ResourceType string
 	Tags         []string
 	TeamID       int
+	TeamName     string
```

Comment:
I don't think we should add `TeamName` to `WorkerSpec`. I think adding it to `ContainerSpec` and using it from there is fine though.

I think adding `TeamName` to `WorkerSpec` will be a bit confusing later on in the worker selection part of the codebase, even though it's not being used in that area right now. `WorkerSpec` contains all the requirements that a worker needs to have in order to be used to run whichever step is executing. With that context, when I read `WorkerSpec.TeamName` I _think_ it represents a worker that belongs to a specific team, which is not what we're trying to convey here as far as usage goes.

From what I can tell there shouldn't be the same confusion with adding `TeamName` to `ContainerSpec`.

Reply:
This makes sense, absolutely. The fact of having already `TeamID` led me to add `TeamName` without thinking about the confusion that could generate.